src/MacroExpander.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * This file contains the “gullet” where macros are expanded
+ * until only non-macro tokens remain.
+ */
+
+var Lexer = require("./Lexer");
+
+function MacroExpander(input, macros) {
+    this.lexer = new Lexer(input);
+    this.macros = macros || {};
+    this.stack = []; // contains tokens in REVERSE order
+    this.discardedWhiteSpace = [];
+}
+
+MacroExpander.prototype.nextToken = function() {
```

Comment:
>  I was thinking it might be clearer to have a separate function that just does the expansion art, thoughts?

I think the function *is* just doing expansion, plus parsing of macro bodies from strings to token arrays. The latter coulc conceivably be moved to a separate function, `getMacroBody` or whatever, if you prefer. I have no strong feelings either way, as long as it's all internal API so we can change it when the need arises.